InfiniteRecharge/src/main/java/frc/robot/subsystem/scoring/shooter/ShooterSubsystem.java

```diff
@@ -184,15 +188,23 @@ public void periodic(float deltaTime) {
     }
 
     public void spinUp() {
+        float targetShooterVelocity = (float) MathUtils
+                .unitConverter(SmartDashboard.getNumber(getName() + "/Shooter Velocity RPM", 60), 600, 8192)
+                / config.shooter.shooterGearRatio;
+
         // Spin up the feeder.
-        feeder.set(
-                SmartDashboard.getNumber(getName() + "/Feeder Output Percent", ShooterConstants.FEEDER_OUTPUT_PERCENT));
-        SmartDashboard.putString(getName() + "/Feeder State", "Feeding");
+        if (ballPropulsionMotor.getSelectedSensorVelocity() >= targetShooterVelocity
+                + config.shooter.feederSpinUpDeadband
+                && ballPropulsionMotor.getSelectedSensorVelocity() <= targetShooterVelocity
+                        - config.shooter.feederSpinUpDeadband) {

```

Comment:
I don't think this logic is quite right. Let's say we want 60 RPMs, and our deadband is 5 RPMs, and our spinner has been spun up to 61 RPM

```
if (61 >= 60 + 5 && 61 <= 60 - 5) {
}
```

which is

```
if (61 >= 65 && 61 <= 55) {
}
```

Both those conditions will never be true.